packages/redux-store/src/recordListener.ts

```diff
@@ -21,7 +26,21 @@ export const removeRecord = createAction<BaseRecord>('record/remove')
  */
 export const startRecordListeners = (agent: Agent, store: Store) => {
   const onDeleted = (event: RecordDeletedEvent<BaseRecord>) => {
-    store.dispatch(removeRecord(event.payload.record))
+    // Extract the record
+    const record = event.payload.record
+    // Retrieve and delete the record if it is of the generic type
+    const retrieveAndDeleteGeneric = async (record: GenRecord) => {
```

Comment:
`GenRecord` is again type casted. It can be a full `BaseRecord` which does not need additional fetching. before retrieving any record from storage, it must be checked whether it is a full record, this happens when `delete` is called or if its just the type and id, this happens when `deleteById` is called.

Reply:
aight. could you try rephrase that? I don't get your point.

Reply:
I'm only retrieving it if it's not a full record/has only id and type. SOrry I'm missing your point I think.

Reply:
`record` in the function interface is now of type `GenRecord` which is just `{id: string, type: string}` which is an unnecessary down-cast of the `BaseRecord` type. Doing this results in having to fetch a record by id for which you already have the record.



```typescript
// pseudo-code

const record: BaseRecord = { ... /* lots of fields */ };

const retrieveAndDeleteRecord = (record: {id: string, type:string }) => {
  // Only call this if the record does not extend `BaseRecord`
  const newRecord = await retrieveRecord(record.id)
  await deleteRecord(newRecord)
}

// if its being called like this, you already have the record ready to be deleted. Fetching it again is unnecessary overhead
retrieveAndDeleteRecord(record)

retrieveAndDeleteRecord({id: 'foo', type: "CredentialExchangeRecord"})

```

Reply:
Thanks!